core/parentchain/block-import-dispatcher/src/triggered_dispatcher.rs

```diff
@@ -100,17 +100,28 @@ where
 		&self,
 		blocks: Vec<SignedBlockType>,
 		events: Vec<RawEventsPerBlock>,
+		is_syncing: bool,
 	) -> Result<()> {
 		let parentchain_id = self.block_importer.parentchain_id();
 		trace!(
-			"[{:?}] Pushing parentchain block(s) and event(s) ({}) ({}) to import queue",
+			"[{:?}] Triggered dispatcher received block(s) and event(s) ({}) ({})",
 			parentchain_id,
 			blocks.len(),
 			events.len()
 		);
-		// Push all the blocks to be dispatched into the queue.
-		self.events_queue.push_multiple(events).map_err(Error::ImportQueue)?;
-		self.import_queue.push_multiple(blocks).map_err(Error::ImportQueue)
+		if is_syncing {
+			trace!(
+				"[{:?}] Triggered is in sync mode, immediately importing blocks and events",
+				parentchain_id
+			);
+			self.block_importer
+				.import_parentchain_blocks(blocks, events)
```

Comment:
I believe this simple bool flag is not enough

* It needs to be clear to other validateers what parentchain block is the first one which must be considered for indirect invocation (in SCV and OCW mode). therefore, that "first relevant parentchain block should be stored in the STF state
* callin it "is_syncing" is misleading. we're always syncing. I think it should be called "is_initial_sync" or similar

Reply:
I agree with all points. I think there was a reason why it has been implemented the way it is now, but I forgot, and I don't see a reason why this is currently like that.

* Substrate introduced the term `is_major_sync`, which is true whenever a node is far behind the best block. This concept does not really apply in our case, as we will send the light-client db to an outdated worker. However, what I am trying to say is that we stay somewhat in the substrate-jargon if we use `is_initial_sync`.

Reply:
Ah, we could also use this as some point maybe for non-authoring sidechain rpc-nodes, if we ever want to have them.